java/memory/src/main/java/io/netty/buffer/ArrowBuf.java

```diff
@@ -366,6 +366,9 @@ public ArrowBuf slice(int index, int length) {
       return this;
     }
 
+    Preconditions.checkPositionIndex(index, this.length);
+    Preconditions.checkPositionIndex(index + length, this.length);
```

Comment:
should we check for length > 0 too ? i.e is a slice of 0 length valid ?

Reply:
In the C++ implementation it is valid to have zero length buffers (they should never be referenced).   I could see going either way.  Is your preference to check for zero length here?

Reply:
I just checked that a zero-sized arrow buffers is valid. so, a slice should be valid too.